contents/docs/error-tracking/_snippets/nextjs-install-error-tracking.mdx

```diff
@@ -117,4 +117,58 @@ export const onRequestError = async (err, request, context) => {
 }
 ```
 
-You can find a full example of both this and client-side error tracking in our [Next.js error monitoring tutorial](/tutorials/nextjs-error-monitoring).
\ No newline at end of file
+You can find a full example of both this and client-side error tracking in our [Next.js error monitoring tutorial](/tutorials/nextjs-error-monitoring).
+
+## Uploading source maps
+
+import CLIDownload from "./cli/download.mdx"
+import CLIAuthenticate from "./cli/authenticate.mdx"
+import CLIInject from "./cli/inject.mdx"
+import CLIUpload from "./cli/upload.mdx"
+
+If your source maps are not publicly hosted, you will need to upload them during your build process to see unminified code in your stack traces. The `posthog-cli` handles this process.
+
+### Download the CLI
+
+<CLIDownload/>
+
+### Authenticate the CLI
+
+<CLIAuthenticate />
+
+### Generate source maps
+
+Next.js disables source maps by default during production builds to prevent you from leaking your source on the client. You can opt-in with the `productionBrowserSourceMaps` configuration flag.
+
+We will set a environment variable to control this so we only generate source maps when building locally (or if you are running a build in CI).
+
+```file=.env.local
+NEXT_PUBLIC_POSTHOG_SOURCE_MAPS=true
+```
+
+Once we have the environment variable set, we can configure Next.js to generate source maps.
+
+```js file=next.config.js 
+/** @type {import('next').NextConfig} */
+const nextConfig = {
+  productionBrowserSourceMaps: process.env.GENERATE_SOURCEMAPS === 'true'
+};
+
+export default nextConfig;
+```
+
+### Inject
+
+Once you've built your application and have bundled assets, inject the context required by PostHog to associate the maps with the served code.
+
+<CLIInject path="./.next" />
+
+### Upload
+
+You will then need to upload the modified assets to PostHog.
+
+<CLIUpload path="./.next" />
+
+You must also ensure that the modified asset bundles uploaded to PostHog are the ones your site serves. If you serve a copy of the bundled assets as they were prior to running `posthog-cli sourcemap inject`, we won't be able to use the uploaded sourcemap to unminify or demangle your stack traces. 
+
+Setting up a CI step to build your app and upload the source maps is a good way to ensure this.
```

Comment:
I feel like we should be more forward in this advice.

Maybe(?): `To ensure that the modified asset bundles uploaded to PostHog are the ones your site serves, set up a CI step to build and upload source maps. If you serve ...`